source/common/upstream/upstream_impl.cc

```diff
@@ -118,11 +119,27 @@ ClusterPtr ClusterImplBase::create(const Json::Object& cluster, ClusterManager&
   if (string_type == "static") {
     new_cluster.reset(new StaticClusterImpl(cluster, runtime, stats, ssl_context_manager));
   } else if (string_type == "strict_dns") {
-    new_cluster.reset(new StrictDnsClusterImpl(cluster, runtime, stats, ssl_context_manager,
-                                               dns_resolver, dispatcher));
+    if (cluster.hasObject("dns_resolvers")) {
+      auto resolvers = cluster.getStringArray("dns_resolvers");
```

Comment:
Just to confirm, you suggest having 'createDnsResolver' take a `std::vector<Network::Address::InstanceConstSharedPtr>`? That sounds good to me.